trade-finance-manager-api/src/v1/rest-mappings/mappings/gef-facilities/mapGefFacilitySnapshot.js

```diff
@@ -44,7 +44,7 @@ const mapGefFacilitySnapshot = (facility, dealSnapshot) => {
 
   facilitySnapshot.facilityProduct = mapFacilityProduct(type);
 
-  facilitySnapshot.facilityStage = mapGefFacilityStage(hasBeenIssued);
+  facilitySnapshot.facilityStage = mapGefFacilityStage(hasBeenIssued, facilityTfm?.facilityStage);
```

Comment:
Bit concerning if everything was passing before without this change 🤔

Reply:
Agreed - it was why I was keen to have this unit test in previously https://github.com/UK-Export-Finance/dtfs2/pull/3770/files/179736077bd2ddc592ab1f8f5cf9b624b869049e#r1816714044 - do you reckon I add it back in?

Reply:
Seperately, it was passing E2E tests as this only impacted non-Gef facilities, and (concerningly) a lot of our E2E tests just check non-GEF facilities. Possibly we should duplicate our E2E deal cancellation tests for GEF facilities? 🤷‍♀️

Reply:
> Seperately, it was passing E2E tests as this only impacted non-Gef facilities, and (concerningly) a lot of our E2E tests just check non-GEF facilities. Possibly we should duplicate our E2E deal cancellation tests for GEF facilities? 🤷‍♀️

Highly recommend adding more tests for non-GEF facilities - sounds like most likely this would have affected non-deal cancellation related tests/scenarios/journeys 😢 

Can we create a ticket in the backlog please?

Reply:
Good idea, I've refactored both `mapGefFacilitySnapshot` and `mapFacilitySnapshot` tests to do this now. Take a look and let me know what you think!
And yes sure, I'll create one